cypress/integration/sign_in.spec.js

```diff
@@ -86,6 +88,131 @@ describe('Sign in flow', () => {
+    });
+
+    it('shows a generic error message when okta api response unknown', function () {
+      cy.visit('/signin');
+      cy.get('input[name="email"]').type('example@example.com');
+      cy.get('input[name="password"]').type('password');
+      cy.mockNext(403, {
+        errorCode: 'E0000147',
+        errorSummary: '',
+        errorLink: '',
+        errorId: '',
+        errorCauses: [],
+      });
+      cy.get('[data-cy=sign-in-button]').click();
+      cy.contains('There was a problem signing in, please try again.');
+    });
+
+    it('loads the redirectUrl upon successful authentication', function () {
+      cy.visit('/signin?returnUrl=https%3A%2F%2Fwww.theguardian.com%2Fabout');
+      cy.get('input[name="email"]').type('example@example.com');
+      cy.get('input[name="password"]').type('password');
+      cy.mockNext(200, {
+        expiresAt: '3000-01-01T00:00:00.000Z',
+        status: 'SUCCESS',
+        sessionToken: 'some-session-token',
+        _embedded: {
+          user: {
+            id: 'okta-id',
+            passwordChanged: '2020-01-01T00:00:00.000Z',
+            profile: {
+              login: 'test.man@example.com',
+              firstName: 'Test',
+              lastName: 'Man',
+              locale: 'en_GB',
+              timeZone: 'Europe/London',
+            },
+          },
+        },
+      });
+
+      // we can't actually check the authorization code flow
+      // so intercept the request and redirect to the guardian about page
+      cy.intercept(
+        `http://localhost:9000/oauth2/customauthserverid/v1/authorize*`,
+        (req) => {
+          req.redirect('https://theguardian.com/about');
+        },
+      );
+
+      cy.get('[data-cy=sign-in-button]').click();
+      cy.contains('About us');
+    });
+
+    it('redirects to the default url if no redirectUrl given', function () {
+      cy.visit('/signin');
+      cy.get('input[name="email"]').type('example@example.com');
+      cy.get('input[name="password"]').type('password');
+      cy.mockNext(200, {
+        expiresAt: '3000-01-01T00:00:00.000Z',
+        status: 'SUCCESS',
+        sessionToken: 'some-session-token',
+        _embedded: {
+          user: {
+            id: 'okta-id',
+            passwordChanged: '2020-01-01T00:00:00.000Z',
+            profile: {
+              login: 'test.man@example.com',
+              firstName: 'Test',
+              lastName: 'Man',
+              locale: 'en_GB',
+              timeZone: 'Europe/London',
+            },
+          },
+        },
+      });
+
+      // we can't actually check the authorization code flow
+      // so intercept the request and redirect to the guardian homepage
+      cy.intercept(
+        `http://localhost:9000/oauth2/customauthserverid/v1/authorize*`,
+        (req) => {
+          req.redirect('https://theguardian.com/');
+        },
+      );
+
+      cy.get('[data-cy=sign-in-button]').click();
+      cy.contains('UK edition');
```

Comment:
It might be worth checking for a different string as I've had issues in the past where Github spins up an instance in a different region causing a different edition to load. Or maybe we just take note of this and deal with it if it happens?

Reply:
Not sure of what other string we could check for? Maybe it's worth checking for an element that always exists instead?

Reply:
Strangely, I've had this problem a few times. There should probably be an element with a `data-cy-test` attribute we can check for

Reply:
Getting back to this PR after so long.. we changed it a while back to look at the URL after the page changed rather than the content.